ref/src/sw_core.F90

```diff
@@ -632,7 +637,15 @@ subroutine d2a2c_vect(sw_corner, se_corner, ne_corner, nw_corner,     &
     if ( is == 1 ) then
       do j = js-1, je+1
         uc(0, j) = c1 * utmp(-2, j) + c2 * utmp(-1, j) + c3 * utmp(0, j)
-        ut(1, j) = edge_interpolate4(ua(-1:2, j), dxa(-1:2, j))
+        ua_slice(1) = ua(-1,j)
+        ua_slice(2) = ua( 0,j)
+        ua_slice(3) = ua( 1,j)
+        ua_slice(4) = ua( 2,j)
+        dxa_slice(1) = dxa(-1,j)
+        dxa_slice(2) = dxa( 0,j)
+        dxa_slice(3) = dxa( 1,j)
+        dxa_slice(4) = dxa( 2,j)
+        ut(1, j) = edge_interpolate4(ua_slice, dxa_slice)
```

Comment:
Did you resolve the warning messages in debug mode to avoid adding this change?

Reply:
Yes!  The warnings in debug mode created 120,000 lines in the log.

Reply:
I suspect the reason is because you modified the `-check` compile options which previously explicitly eliminated the checking for array temporaries.  This code generates a ton of temporary arrays due to the passing of slices to the subroutine.  Your update works fine to eliminate the temporary arrays, but it induces a change to the code, that I'd prefer to avoid if possible.  Please check to see whether the 120,000 lines of warnings can be eliminated by restoring the original `-check` options.

Reply:
I backed out my changes.